internal/pkg/service/common/etcdop/mutex_test.go

```diff
@@ -187,9 +187,9 @@ func TestMutex_ParallelWork(t *testing.T) {
 					// Create client and session
 					client := etcdhelper.ClientForTest(t, etcdCfg)
 					session, errCh := NewSessionBuilder().Start(ctx, sessionWg, logger, client)
+					readyWg.Done()
 					require.NotNil(t, session)
 					require.NoError(t, <-errCh)
-					readyWg.Done()
```

Comment:
The problem probably arose here, `require.*` terminates the goroutine in case of an error, so `wg` is not unlocked.

The original issue was probably a network error, we'll see when it happens again. The timeout should be now solved.